universum/modules/vcs/github_actions_vcs.py

```diff
@@ -0,0 +1,126 @@
+            """)
+
+        self.payload: str = self.read_and_check_multiline_option("payload", """
+            GitHub web-hook payload JSON is not specified.
+
+            Please pass incoming web-hook request payload to this parameter directly via
+            '--github-payload' ('-ghp') command line parameter or by setting GITHUB_PAYLOAD
+            environment variable, or by passing file path as the argument value (start
+            filename with '@' character, e.g. '@/tmp/file.json' or '@payload.json' for
+            relative path starting at current directory). Please note, that when passing
+            a file, it's expected to be in UTF-8 encoding.
+            """)
+
+        try:
+            self.payload_json: dict = json.loads(self.payload)
+            self.settings.repo = self.payload_json['repository']['html_url']
+            self.settings.refspec = self.payload_json['pull_request']['head']['ref']
+        except json.decoder.JSONDecodeError as error:
+            self.error(f"Provided payload value could not been parsed as JSON "
+                       f"and returned the following error:\n {error}")
+
+    def _clone(self, history_depth: int, destination_directory: str, clone_url: str) -> None:
+        parsed_repo: urllib.parse.SplitResult = urllib.parse.urlsplit(clone_url)
+        if parsed_repo.scheme == "https" and not parsed_repo.username:
+            new_netloc = f"{self.settings.token}@{parsed_repo.netloc}"
+            parsed_repo = (parsed_repo.scheme, new_netloc, parsed_repo.path,
+                           parsed_repo.query, parsed_repo.fragment)  # type: ignore
+        clone_url = str(urllib.parse.urlunsplit(parsed_repo))
+        super()._clone(history_depth, destination_directory, clone_url)
+
+    def code_review(self) -> Self:
+        self.reporter = self.reporter_factory()
+        self.reporter.subscribe(self)
+        return self
+
+    def update_review_version(self) -> None:
+        self.out.log("GitHub has no review versions")
+
+    def get_review_link(self) -> str:
+        return self.payload_json['pull_request']['html_url']
+
+    def is_latest_version(self) -> bool:
+        return True
+
+    def _report(self, url, request: dict) -> None:
+        headers: Dict = {
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer { self.settings.token }"
+        }
+
+        utils.make_request(url, request_method="POST", json=request, headers=headers, timeout=5*60)
+
+    def code_report_to_review(self, report: dict) -> None:
+        # git show returns string, each file separated by \n,
+        # first line consists of commit id and commit comment, so it's skipped
+        commit_files: List[str] = self.repo.git.show("--name-only", "--oneline",
+                                                     self.settings.checkout_id).split('\n')[1:]
+        # NB! When using GITHUB_TOKEN, the rate limit is 1,000 requests per hour per repository.
```

Comment:
It is not clear why this comment is here. It doesn't seem to relate to any code around.

Reply:
Added explanation.